src/controllers/commentController.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable class-methods-use-this */
+import Response from '../utils/response';
+import CommentService from '../repositories/commentRepository';
+import UserService from '../repositories/userRepository';
+import RequestService from '../repositories/requestRepository';
+
+class CommentController {
+  /**
+   * Creates a new comment.
+   * @param {object} req request
+   * @param {object} res response
+   * @param {object} next next
+   * @returns {object} response object
+   */
+  static async addComment(req, res, next) {
+    try {
+      const { comment } = req.body;
+      const id = Number(req.params.id);
+      const requestExists = await RequestService.findRequestById(id);
+      if (!requestExists) {
+        const response = new Response(res, 404, 'Request is not found to add a comment');
+        return response.sendErrorMessage();
+      }
+      const user = await UserService.findByUserId(req.userData.id);
+      const addedComment = await CommentService.addComment({
+        user_id: user.id,
+        request_id: id,
+        comment
+      }, {
+        fields: ['user_id', 'request_id', 'comment']
+      });
+      const response = new Response(res, 201, 'comment sucessfully created', addedComment);
+      response.sendSuccessResponse();
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * gets all comments by request
+   * @param {object} req request
+   * @param {object} res response
+   * @param {object} next next
+   * @returns {object} response
+   */
+  static async getCommentsByRequest(req, res, next) {
+    try {
+      const id = Number(req.params.id);
+      const requestExists = await RequestService.findRequestById(id);
+      if (!requestExists) {
+        const response = new Response(res, 404, 'Request is not found with this ID');
+        return response.sendErrorMessage();
+      }
+      const comments = await CommentService.getRequestCommentById(req.params.id);
+      return res.status(200).json({ status: 200, message: 'Comments fetched successfully', comments });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * delete comment by request
+   * @param {object} req request
+   * @param {object} res response
+   * @param {object} next
+   * @returns {object} object response
+   */
+
+  static async deleteComment(req, res, next) {
+    try {
+      const id = Number(req.params.id);
+      const commentFound = await CommentService.getCommentById(id);
+      if (!commentFound) {
+        const response = new Response(res, 404, 'Comment not found');
+        return response.sendErrorMessage();
+      }
+      await CommentService.deleteComment(id);
```

Comment:
what is the deleteComment didn't work? can I recommend you to use then here. because it is possible that you can lie to the user that the  comment was deleted yet it failed. unless you convince me that your error handling is perfect